cmake/cxxflags_common.cmake

```diff
@@ -0,0 +1,80 @@
+
+
+# This is unfortunately still needed to disable exceptions/RTTI since modern CMake still has no builtin support...
+# E.g. replace_cxx_flag("/EHsc", "/EHs-c-")
+macro(replace_cxx_flag pattern text)
+    foreach (flag
+        CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_RELEASE
+        CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO)
+
+        string(REGEX REPLACE "${pattern}" "${text}" ${flag} "${${flag}}")
+
+    endforeach()
+endmacro()
+
+# Fixup default compiler settings
+if (MSVC)
```

Comment:
`if (MSVC)`, I guess this must be the way CMake identifies the microsoft compiler.